src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -254,27 +255,30 @@ public void setSheetPos(String sheetName, int pos) {
         workbook.setSheetOrder(sheetName, pos);
     }
 
-    public String[] getDefinedNames(boolean validOnly) {
-        //TODO getNameAt is deprecated in POI 4.0 -
-        // "New projects should avoid accessing named ranges by index."
-        int count = workbook.getNumberOfNames();
-        // String[] nameNames = new String[count];
-        ArrayList<String> nameNames = new ArrayList<String>();
-
-        for(int i = 0; i < count; i++) {
-            Name namedRegion = workbook.getNameAt(i);
-            // if valid only, check corresponding reference formula validity
-            if(validOnly && !isValidNamedRegion(namedRegion)) continue;
-            nameNames.add(namedRegion.getNameName());
-        }
+    public String[] getDefinedNames(boolean validOnly, String worksheetScope) {
```

Comment:
I guess there is still a "global"/"any" scope?

Reply:
Given some details below I infer that `worksheetScope = null` is the "globa"/"any" scope ...

Reply:
clarification (see https://github.com/miraisolutions/xlconnect/pull/146#issuecomment-923745089 on the R side): `worksheetScope = null` means _any_, `worksheetScope = ""` means global scope _only_